File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashtable={}
        for i in range(len(nums)):   
            hashtable[nums[i]]=i
        for i in range(len(nums)):
            if target-nums[i] in hashtable:
                if hashtable[target-nums[i]] != i:
                    return [i, hashtable[target-nums[i]]]
        return []
        
        
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mapNumber;
        vector<int> answer;
        for(int i =0;i < nums.size();i++){
            if(mapNumber.find(target-nums[i])!=mapNumber.end()){
                
                answer.push_back(mapNumber.find(target-nums[i])->second);
                answer.push_back(i);
                break;
            }
            mapNumber[nums[i]] = i;
        }
        return answer;
    }
};